### DISCUSSION TITLE
Feedback-based changes for v1 (request for comments)

### DISCUSSION THREAD

**Discussion by xPi2** [nation3/court#51] [Ideas]
A week ago, I was at DevCon and had the opportunity to share the Nation3 Court vision with great developers and founders. I gathered some feedback from those discussions that led me to propose the following changes.

## Support N3 non-native tokens

From the beginning, I had the gut feeling that we would need to extend the token support in Nation3 Agreements to more than $NATION. We discussed the possibility of implementing other N3-issued tokens to provide more stable assets for the agreements, different collateralization strategies, etc.

I'm confident now that we need to support any token users want to use as collateral on the agreements. Without this, our potential as a dispute resolution service for more than Nation3DAO would be minimal.

The main problem I had with this approach is that it conflicts with our idea of being able to sustain the system based on the usage of the jurisdiction rather than on the arising of disputes. With the restriction to N3 native tokens, the DAO could use the demand over those tokens to sustain itself. But, with external assets, we would need to introduce protocol fees or other "taxation" models that people didn't like. Here comes the second point.

## Dispute on staked native token

In the current implementation, the collateral token and dispute token are the same, and the dispute fee is paid when a dispute arises. Even if we introduce support to use any compatible token as collateral, we should keep a native token (NATION) as the dispute fee token. This could be the solution to the sustainability problem previously described without having to charge fees for the usage of the agreement frameworks.

Going a step further, instead of relying on disputes to attract demand over the native tokens (we don't want those incentives in place), I propose we require staking the dispute token to join agreements. To do so, you would deposit NATION into a stake that you would lock during the agreement duration. If you raise a dispute, the dispute fee will be drawn from your stake, and whenever an agreement is finalized without dispute, you will continue using your staked tokens on further agreements. This same stake mechanism could be used for appeals, so the appeal fee is only charged when your appeal is rejected, making fair appeals accessible while keeping the deterrent for abusing the appeals.

This staking mechanism could be further extended to create incentives for good long-term use of the jurisdiction, and also be used between diferent N3 services. But I will only go deeper once we can iterate based on actual usage data. 

---

These changes would be optional in the upcoming demo version of the system but can be introduced for the first official release if the feedback is positive.

**Comment by AaronAbuUsama** [comment]
When you first walked me through the system it was clear that without the option of at least one stablecoin we would be handicapping the court. I am very happy this is the direction we are moving in. Also, I agree in principle that fees should be paid in `$NATION` for the reasons you mentioned, however, if we do this at the protocol level it would require users to then acquire another token which they may not have made the UX worse. either the fee token should be converted automatically or done by the DAO itself. I have some other comments about the fee dynamics but ill save that for the relevant thread

**Reply by xPi2** [reply]
So your suggestion is that the user stake the dispute fee in the same token as the collateral? If that's the case this would break the idea of being able to capitalize on the use of the jurisdiction instead of the disputes, as the DAO would only capture value when the dispute fee is paid.

Other optión is to solve this at UI level, allowing the user to buy (or stake) the required staked NATION at the moment of joining an agreement with the token they want, so at protocol level things keep simpler (and cheaper). Because once you have your "staked" NATION you can continue to use it in multiple agreements.

**Reply by luisivan** [reply]
This would be ideal. Stake once, use everywhere. Also I like the time decay model, so that you'd need to top up your stake over time. It generates constant demand, instead of one-off.

**Reply by luisivan** [reply]
While writing this I realized I'm describing a passport 😂 I guess they will both converge soon

**Reply by AaronAbuUsama** [reply]
this works even better! User collateralizes with `USDC` --> fee is converted into `NATION` and staked. User gets more utility, `NATION` gets more demand. win-win

**Reply by luisivan** [reply]
I guess the problem with that is volatility though...

**Comment by okhaimie-dev** [comment]
Yes, I like the idea of opening Nation3 court to accept non native ERC-20 token as collateral. 

I am also of the opinion that most non crypto natives don't like to pay fees so this might not be the best route if we plan to scale adoption, especially ( + ETH mainnet gas fee ), but it might also be a condition that only non-citizens pay a fee which might be a driver for people to buy our regular passport depending on the cost and our marketing strategies. ( Although, I must emphasize that including a fee might be a barrier to the Court adoption for non Nation3 members. )


===

I love your suggestion on staking dispute token and reuse for future agreements. Seems like that is the best path forward for the options we have.

**Reply by xPi2** [reply]
Your need to pay at least the arbitration fee, arbitrators need to be paid in some way, if not this service wouldn't be sustainable. Said that, we can explore incentives for Nation3 citizenship that can generate synergies between both.
